components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java

```diff
@@ -1565,4 +1569,88 @@ public List<API> getAllApisByLabel(String gatewayLabel, Boolean expand) {
+            throws APIManagementException {
+        Map<String, Object> subscriptionDetails = new HashMap<>();
+        int subscriptionId = -1;
+        String subscriptionUUID = UUID.randomUUID().toString();
+
+        String checkDuplicateQuery = SQLConstants.CHECK_EXISTING_SUBSCRIPTION_API_SQL;
+        String sqlQuery = SQLConstants.ADD_SUBSCRIPTION_SQL;
+
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            try (PreparedStatement ps = connection.prepareStatement(checkDuplicateQuery)) {
+                ps.setInt(1, apiId);
+                ps.setInt(2, appId);
+                try (ResultSet resultSet = ps.executeQuery()) {
+                    //If the subscription already exists
+                    if (resultSet.next()) {
+                        String subStatus = resultSet.getString("SUB_STATUS");
+                        String subCreationStatus = resultSet.getString("SUBS_CREATE_STATE");
+                        if ((APIConstants.SubscriptionStatus.UNBLOCKED.equals(subStatus) ||
+                                APIConstants.SubscriptionStatus.ON_HOLD.equals(subStatus) ||
+                                APIConstants.SubscriptionStatus.REJECTED.equals(subStatus)) &&
+                                APIConstants.SubscriptionCreatedStatus.SUBSCRIBE.equals(subCreationStatus)) {
+
+                            throw new SubscriptionAlreadyExistingException(
+                                    String.format("Subscription already exists for API/API Prouct %s in Application %s",
+                                            apiId, appId));
+                        }
+                    }
+                }
+            }
+
+            String subscriptionIDColumn = "SUBSCRIPTION_ID";
+            if (connection.getMetaData().getDriverName().contains("PostgreSQL")) {
+                subscriptionIDColumn = "subscription_id";
+            }
+            try (PreparedStatement preparedStForInsert = connection.prepareStatement(sqlQuery,
+                    new String[]{subscriptionIDColumn})) {
+                preparedStForInsert.setString(1, tier);
+                preparedStForInsert.setString(10, tier);
+                preparedStForInsert.setInt(2, apiId);
+                preparedStForInsert.setInt(3, appId);
+                preparedStForInsert.setString(4, APIConstants.SubscriptionStatus.UNBLOCKED);
+                preparedStForInsert.setString(5, APIConstants.SubscriptionCreatedStatus.SUBSCRIBE);
+                preparedStForInsert.setString(6, subscriber);
+
+                Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+                preparedStForInsert.setTimestamp(7, timestamp);
+                preparedStForInsert.setTimestamp(8, timestamp);
+                preparedStForInsert.setString(9, subscriptionUUID);
+
+                preparedStForInsert.executeUpdate();
+                try (ResultSet rs = preparedStForInsert.getGeneratedKeys()) {
```

Comment:
use 'connection.setAutoCommit(false);' and  'connection.commit();'

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12539/commits/9cd326adf51117359068bd6457efdaf33d209aea